spec.md

```diff
@@ -572,217 +572,152 @@ See [@sec:mem] for more details.
 ```
 
+Vectors with length 0 are permitted.
+For details on how 0-length vectors are lowered, see the FIRRTL ABI Specification.
+
 ### Bundle Types
 
-A bundle type is used to express a collection of nested and named types.
-All fields in a bundle type must have a given name, and type.
-All names must be legal identifiers.
+A bundle type is used to represent a collection of values.
+They can also be used to facilitate bidirectional connections between circuit components.
+
+A bundle type consists of zero or more fields.
+Each field has a name and a type.
+Fields may also be flipped.
 
 The following is an example of a possible type for representing a complex number.
 It has two fields, `real`{.firrtl}, and `imag`{.firrtl}, both 10-bit signed integers.
 
-``` firrtl
-{real: SInt<10>, imag: SInt<10>}
-```
-
-Additionally, a field may optionally be declared with a *flipped* orientation.
-
-``` firrtl
-{word: UInt<32>, valid: UInt<1>, flip ready: UInt<1>}
+```firrtl
+{ real : SInt<10>, imag : SInt<10> }
 ```
 
-In a connection between circuit components with bundle types, the data carried by the flipped fields flow in the opposite direction as the data carried by the non-flipped fields.
+The types of each field may be any type, including other aggregate types.
 
-As an example, consider a module output port declared with the following type:
+```firrtl
+{ real : { word : UInt<32>, valid : UInt<1>, flip ready : UInt<1> },
+  imag : { word : UInt<32>, valid : UInt<1>, flip ready : UInt<1> } }
 
-``` firrtl
-output a: {word: UInt<32>, valid: UInt<1>, flip ready: UInt<1>}
 ```
 
-In a connection to the `a`{.firrtl} port, the data carried by the `word`{.firrtl} and `valid`{.firrtl} sub-fields will flow out of the module, while data carried by the `ready`{.firrtl} sub-field will flow into the module.
-More details about how the bundle field orientation affects connections are explained in [@sec:connects].
-
-As in the case of vector types, a bundle field may be declared with any type, including other aggregate types.
-
-``` firrtl
-{real: {word: UInt<32>, valid: UInt<1>, flip ready: UInt<1>},
- imag: {word: UInt<32>, valid: UInt<1>, flip ready: UInt<1>}}
+Here is an example of a bundle with a flipped field.
+Because the `ready` field is marked with the keyword `flip`, it will indicate the flow will be opposite of the `word` and `valid` fields.
 
+```firrtl
+{ word : UInt<32>, valid : UInt<1>, flip ready : UInt<1> }
 ```
 
-When calculating the final direction of data flow, the orientation of a field is applied recursively to all nested types in the field.
-As an example, consider the following module port declared with a bundle type containing a nested bundle type.
+As an example of how `flip`{.firrtl} works in context, consider a module declared like this:
 
-``` firrtl
-output myport: {a: UInt, flip b: {c: UInt, flip d: UInt}}
+```firrtl
+module Processor :
+  input enq : { word : UInt<32>, valid : UInt<1>, flip ready : UInt<1> }
+  ; ...
```

Comment:
Nit: the ellipsis is unnecessary. You could make it clearly irrelevant by making `Processor` an `extmodule`.